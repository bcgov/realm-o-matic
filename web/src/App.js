import React, { Component } from 'react';
import Helmet from 'react-helmet';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Helmet>
          <meta charSet="utf-8" />
          <title>Realm-o-Matic</title>
          <meta name="description" content="Self-service App for KeyCloak Realm Provosioning" />
        </Helmet>
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            hello!!1
          </a>
        </header>
      </div>
    );
  }
}

export default App;
