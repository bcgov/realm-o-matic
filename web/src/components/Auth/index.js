export { default as AuthModal } from './AuthModal';
export { LoginRoute } from './LoginRoute';
export { default as ProtectedRoute } from './ProtectedRoute';
