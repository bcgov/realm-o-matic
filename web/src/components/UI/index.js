export { default as Header } from './Header';
export { Footer } from './Footer';
export { Layout } from './Layout';
