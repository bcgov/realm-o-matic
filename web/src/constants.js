//
// Realm-o-Matic
//
// Copyright Â© 2019 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Shelly Xue Han
//

import config from './config.json';

export const SSO_CONFIG = {
  baseURL: process.env.REACT_APP_SSO_BASE_URL || config.ssoBaseUrl,
  realmName: process.env.REACT_APP_SSO_REALM_NAME || config.ssoRealmName,
  clientId: process.env.REACT_APP_SSO_CLIENT_ID || config.ssoClientId,
};

export const API = {
  BASE_URL: () =>
    process.env.NODE_ENV === 'development'
      ? 'http://localhost:8000/api/v1/'
      : `${window.location.origin}/api/v1/`,
  TIME_OUT: 40000,
  IDP: () => 'idps',
};

export const TEST_IDS = {
  APP: {
    LOGIG: 'login-button',
    GET_IDPS: 'get-idps-button',
  },
};

export const APP_INFO = {
  NAME: 'Realm-o-Matic',
};

export const SSO_IDP = {
  GITHUB: 'github',
  IDIR: 'idir',
};

export const FOOTER_LINKS = [
  {
    name: 'Home',
    link: '.',
  },
  {
    name: 'Disclaimer',
    link: 'https://www2.gov.bc.ca/gov/content/home/disclaimer',
  },
  {
    name: 'Privacy',
    link: 'https://www2.gov.bc.ca/gov/content/home/privacy',
  },
  {
    name: 'Accessibility',
    link: 'https://www2.gov.bc.ca/gov/content/home/accessibility',
  },
  {
    name: 'Copyright',
    link: 'https://www2.gov.bc.ca/gov/content/home/copyright',
  },
  {
    name: 'Contact Us',
    link: 'https://github.com/bcgov/realm-o-matic',
  },
];
