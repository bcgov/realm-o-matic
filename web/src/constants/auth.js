//
// Realm-o-Matic
//
// Copyright Â© 2019 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Shelly Xue Han
//

import config from '../config.json';

export const SSO_CONFIG = {
  baseURL: process.env.REACT_APP_SSO_BASE_URL || config.ssoBaseUrl,
  realmName: process.env.REACT_APP_SSO_REALM_NAME || config.ssoRealmName,
  clientId: process.env.REACT_APP_SSO_CLIENT_ID || config.ssoClientId,
};

export const SSO_IDP = {
  GITHUB: 'github',
  IDIR: 'idir',
};

export const ACCESS_CONTROL = {
  REQUESTER_ROLE: 'devhub_kc_requester',
  REVIEWER_ROLE: 'devhub_kc_reviewer',
  NO_ROLE: 'unauthorized',
};
