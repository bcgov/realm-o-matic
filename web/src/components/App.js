import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Button, List } from 'semantic-ui-react';
import styled from '@emotion/styled';
import implicitAuthManager from '../utils/auth';
import { getIdps } from '../actionCreators';
import { authenticateFailed, authenticateSuccess } from '../actions';
import { TEST_IDS } from '../constants';

const StyledApp = styled.div`
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

export class App extends Component {
  componentDidMount = () => {
    implicitAuthManager.registerHooks({
      onAuthenticateSuccess: () => this.props.login(),
      onAuthenticateFail: () => this.props.logout(),
    });
    // don't call function if on localhost
    if (!window.location.host.match(/localhost/)) {
      implicitAuthManager.handleOnPageLoad();
    }
    try {
      const iamId = implicitAuthManager.idToken.data.sub;
      const token = implicitAuthManager.idToken.bearer;
    } catch (err) {
      console.log('---implicitAuthManager----not logged in');
    }
  };

  render() {
    const onClick = () => {
      this.props.getIdps();
    };

    const onClickSSO = () => {
      window.location = implicitAuthManager.getSSOLoginURI();
    };

    const idps = this.props.idps ? this.props.idps : [];
    const buttons = !this.props.userId ? (
      <Button onClick={onClickSSO} data-testid={TEST_IDS.APP.LOGIG}>
        sso login
      </Button>
    ) : (
      <Button onClick={onClick} data-testid={TEST_IDS.APP.GET_IDPS}>
        get idps
      </Button>
    );

    return (
      <StyledApp>
        <p>Welcome!</p>
        {buttons}
        <p>{this.props.errorMessage}</p>
        <List items={idps} />
      </StyledApp>
    );
  }
}

const mapStateToProps = state => {
  return {
    isAuthenticated: state.authentication.isAuthenticated,
    email: state.authentication.email,
    userId: state.authentication.userId,
    errorMessage: state.getIdps.errorMessage,
    idps: state.getIdps.idps,
    getIdpsStarted: state.getIdps.getIdpsStarted,
  };
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      login: () => dispatch(authenticateSuccess()),
      logout: () => dispatch(authenticateFailed()),
      getIdps,
    },
    dispatch
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
