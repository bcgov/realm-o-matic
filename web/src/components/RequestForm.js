import React from 'react';
import * as Survey from 'survey-react';
import 'survey-react/survey.css';
import styled from '@emotion/styled';
import { TEST_IDS } from '../constants/ui';

const StyledForm = styled.div`
  text-align: left;

  .form-body {
    border-top: 2px solid #003366 !important;
  }

  .BC-Gov-PrimaryButton {
    background-color: #003366 !important;
    font-size: 1em !important;
    border-radius: 4px !important;
  }

  .BC-Gov-SecondaryButton {
    font-size: 1em !important;
    background: none !important;
    border-radius: 4px !important;
    border: 1px solid #003366 !important;
    color: #003366 !important;
  }

  .form-title {
    font-size: 1.5em !important;
  }

  .sv_progress_bar {
    background-color: #003366 !important;
  }

  input[type='text']:disabled {
    cursor: not-allowed;
  }
`;

// survey-react classes:
const formCss = {
  body: 'sv_body form-body',
  pageTitle: 'sv_page_title form-title',
  navigation: {
    complete: 'btn sv_complete_btn BC-Gov-PrimaryButton',
    prev: 'btn sv_prev_btn BC-Gov-SecondaryButton',
    next: 'btn sv_next_btn BC-Gov-SecondaryButton',
    start: 'btn sv_start_btn BC-Gov-PrimaryButton',
  },
};

export const RequestForm = ({ formModal, initialInfo, isDisplayMode, onComplete }) => {
  const completeAction = onComplete => {
    // TODO: scroll to top
    onComplete();
  };
  // setup survey form:
  const requestForm = new Survey.Model(formModal);
  requestForm.data = initialInfo;
  requestForm.mode = isDisplayMode ? 'display' : 'write';

  return (
    <StyledForm data-testid={TEST_IDS.REQUEST.FORM}>
      <Survey.Survey model={requestForm} css={formCss} onComplete={completeAction} />
    </StyledForm>
  );
};
