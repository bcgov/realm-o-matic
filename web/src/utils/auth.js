import { ImplicitAuthManager } from '@bcgov/common-web-utils';
import { SSO_CONFIG } from '../constants/auth';

const iam = new ImplicitAuthManager(SSO_CONFIG);

export default iam;
