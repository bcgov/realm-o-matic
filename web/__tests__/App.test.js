import React from 'react';
import { render } from 'react-testing-library';
import { App } from '../src/components/App';
import { ACCESS_CONTROL } from '../src/constants/auth';

jest.mock('../src/components/UI/Layout.js', () => () => <div />);

describe('App Component', () => {
  const defaultProps = {
    authentication: {
      isAuthenticated: true,
    },
    authorization: {
      authCode: ACCESS_CONTROL.REVIEWER_ROLE,
    },
    getRequests: {},
    newRequest: {},
  };

  it('renders without crashing', () => {
    const { container } = render(<App {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
