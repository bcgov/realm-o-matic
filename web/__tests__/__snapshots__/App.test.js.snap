// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component displays after logged in 1`] = `
<div
  class="css-bpit-StyledApp eamkuan0"
>
  <p>
    Welcome!
  </p>
  <button
    class="ui button"
    data-testid="get-idps-button"
  >
    get idps
  </button>
  <p />
  <div
    class="ui list"
    role="list"
  />
</div>
`;

exports[`App Component provide list of idps 1`] = `
<div
  class="css-bpit-StyledApp eamkuan0"
>
  <p>
    Welcome!
  </p>
  <button
    class="ui button"
    data-testid="login-button"
  >
    sso login
  </button>
  <p />
  <div
    class="ui list"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      abc
    </div>
    <div
      class="item"
      role="listitem"
    >
      def
    </div>
  </div>
</div>
`;

exports[`App Component renders without crashing 1`] = `
<div
  class="css-bpit-StyledApp eamkuan0"
>
  <p>
    Welcome!
  </p>
  <button
    class="ui button"
    data-testid="login-button"
  >
    sso login
  </button>
  <p />
  <div
    class="ui list"
    role="list"
  />
</div>
`;
